File: 0704/Java/Soln_2.java
class Solution {
    public int search(int[] nums, int target) {
        // [l, h) approach : Find the insertion Index
        // if the insertion index = size(nums) or nums[insertion_index]!= target return not found
        int low = 0;
        int high = nums.length;
        while(low < high){
            int mid = low + ((high - low)>>1);
            if(nums[mid] >= target)
                high = mid;
            else
                low = mid + 1;
        }
        return (low==nums.length || nums[low]!=target) ? -1:low;
    }
}

File: 0169/Cpp/soln_2.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int candidate = 0;
        for(int i=0,votes=0;i<nums.size();i++){
            if(!votes)
                votes=1, candidate=nums[i];
            else
                votes+= candidate==nums[i] ? 1:-1;
        }
        return candidate;
    }
};